dl-validator-data/src/near.rs

```diff
@@ -0,0 +1,159 @@
+    }
+}
+
+impl Protocol for NearPrototocl {
+    /// Downloads Near validator data via the [`validators`] RPC method.
+    ///
+    /// [`validators`]: https://docs.near.org/api/rpc/network#validation-status
+    fn download_validator_data(&self) -> anyhow::Result<Vec<ValidatorData>> {
+        // Construct parameters for the HTTP request to RPC.
+        let params = serde_json::json!({
+            "jsonrpc": "2.0",
+            "method": "validators",
+            "id": "dontcare",
+            // If `self.block.is_none()` this serializes to `[null]` and latest block is queried.
+            "params": vec![self.block],
+        });
+
+        // Construct and send the HTTP request.
+        let client = reqwest::blocking::Client::new();
+        let res = client.post(&self.rpc_url).json(&params).send()?;
+
+        // Handle HTTP and RPC errors.
+        anyhow::ensure!(
+            res.status() == 200,
+            "expected HTTP status code 200, got {} with response body\n\t{}",
+            res.status(),
+            res.text()?,
+        );
+        let rpc_res: RpcResponse = res.json()?;
+        if let Some(err) = rpc_res.error {
+            anyhow::bail!("rpc error: {}", err);
+        }
+
+        let validators = rpc_res
+            .result
+            .expect("response without error should have a result")
+            .current_validators
+            .into_iter()
+            .map(|v| v.into())
+            .collect();
+
+        Ok(validators)
+    }
+}
+
+/// The expected response for the `validators` RPC method. This struct contains only fields used in
+/// this module. Additional fields are documented in Near RPC [docs].
+///
+/// [docs]: https://docs.near.org/api/rpc/network#validation-status
+#[derive(Deserialize, Debug)]
+struct RpcResponse {
+    result: Option<RpcResult>,
+    /// Near RPC returns some errors in the response body of a 'successful' (status code 200) HTTP
+    /// request.
+    error: Option<serde_json::Value>,
+}
+
+#[derive(Deserialize, Debug)]
+struct RpcResult {
+    current_validators: Vec<RpcValidatorData>,
+}
+
+/// Data per validator returned by the RPC. This struct contains only fields used in
+/// this module. Additional fields are documented in Near RPC [docs].
+///
+/// [docs]: https://docs.near.org/api/rpc/network#validation-status
+#[derive(Deserialize, Debug)]
+struct RpcValidatorData {
+    account_id: String,
+    /// String representation of `u128`.
+    stake: String,
+}
+
+impl From<RpcValidatorData> for ValidatorData {
+    fn from(data: RpcValidatorData) -> Self {
+        Self {
+            account_id: data.account_id,
+            stake: data
+                .stake
+                .parse::<u128>()
+                .expect("should parse stake as u128"),
+            is_malicious: None,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::protocol::Protocol;
+
+    use super::NearPrototocl;
+
+    const RPC_URL: &str = "https://rpc.testnet.near.org";
```

Comment:
Since the block height is fixed you might need to use `https://archival-rpc.testnet.near.org` instead because eventually the block will be too old for non-archival nodes to know about.

Reply:
Addressed in 8fc60a1.